src/groupeddataframe/splitapplycombine.jl

```diff
@@ -823,12 +834,15 @@ function select(@nospecialize(f::Base.Callable), gd::GroupedDataFrame; copycols:
     return select(gd, f, copycols=copycols, keepkeys=keepkeys, ungroup=ungroup)
 end
 
-select(gd::GroupedDataFrame, @nospecialize(args::Union{Pair, Base.Callable, ColumnIndex, MultiColumnIndex,
-                                                       AbstractVecOrMat}...);
-       copycols::Bool=true, keepkeys::Bool=true, ungroup::Bool=true, renamecols::Bool=true) =
-    _combine_prepare(gd, Ref{Any}(map(x -> broadcast_pair(parent(gd), x), args)),
-                     copycols=copycols, keepkeys=keepkeys,
-                     ungroup=ungroup, keeprows=true, renamecols=renamecols)
+function select(gd::GroupedDataFrame, @nospecialize(args::Union{Pair, Base.Callable, ColumnIndex,
+                                                                MultiColumnIndex, AbstractVecOrMat}...);
+                copycols::Bool=true, keepkeys::Bool=true, ungroup::Bool=true, renamecols::Bool=true)
+    res = _combine_prepare(gd, Ref{Any}(map(x -> broadcast_pair(parent(gd), x), args)),
+                           copycols=copycols, keepkeys=keepkeys,
+                           ungroup=ungroup, keeprows=true, renamecols=renamecols)
+    copycols || _dealias_dataframe(parent(res))
```

Comment:
This isn't needed for `select!`?

Reply:
As you noted - this is a complex code. It is not needed because `select!` calls `select` internally. Same with `transform!`.